python/3.12/Dockerfile

```diff
@@ -112,22 +112,7 @@ RUN \
 ARG PIP_VERSION
 
 RUN set -ex; \
-    \
-    apk add --no-cache --virtual .fetch-deps openssl; \
-    \
-    curl -L -o get-pip.py 'https://bootstrap.pypa.io/get-pip.py'; \
-    \
-    apk del .fetch-deps; \
-    \
-    python get-pip.py \
-        --disable-pip-version-check \
-        --no-cache-dir \
-        pip==${PIP_VERSION} \
-    ; \
-    pip --version; \
-    \
-    find /usr/local -depth \
-        \( \
-            -type d -a \( -name test -o -name tests \) \
-        \) -exec rm -rf '{}' +; \
-    rm -f get-pip.py
+    python -m ensurepip --upgrade; \
+    pip3 install --no-cache-dir --upgrade pip=="${PIP_VERSION}"; \
+    ln -s /usr/local/bin/pip3 /usr/local/bin/pip; \
```

Comment:
Can we maybe get rid of this by using `--default-pip`?

(from https://github.com/docker-library/python/issues/951#issuecomment-2307759828).

Reply:
I'm blind :see_no_evil: Sure, done!